File: contract/src/bounty.rs
use soroban_sdk::{contracttype, Address, String};

#[contracttype]
#[derive(Clone, PartialEq, Debug)]
pub enum BountyStatus {
    Open,
    InProgress,
    Submitted,
    Completed,
    Disputed,
    Refunded,
}

#[contracttype]
#[derive(Clone, Debug)]
pub struct Bounty {
    pub id: u64,
    pub poster: Address,
    pub hunter: Option<Address>,
    pub title: String,
    pub description: String,
    pub reward: i128, // in stroops (1 XLM = 10_000_000)
    pub work_url: Option<String>,
    pub status: BountyStatus,
    pub created_at: u64,
    pub submitted_at: Option<u64>,
    pub deadline_hours: u64, // default 72
}

File: src/types/bounty.ts
// TypeScript types mirroring the Soroban contract structs

export enum BountyStatus {
  Open = 'Open',
  InProgress = 'InProgress',
  Submitted = 'Submitted',
  Completed = 'Completed',
  Disputed = 'Disputed',
  Refunded = 'Refunded',
}

export interface Bounty {
  id: bigint
  poster: string
  hunter: string | null
  title: string
  description: string
  reward: bigint // stroops (1 XLM = 10_000_000 stroops)
  work_url: string | null
  status: BountyStatus
  tags: string[]
  created_at: bigint
  submitted_at: bigint | null
  deadline_hours: bigint
}

export interface Reputation {
  wallet: string
  completed: number
}

export type TransactionStatus = 'idle' | 'pending' | 'confirmed' | 'failed'

export interface TxState {
  status: TransactionStatus
  hash?: string
  error?: string
}
